### PR TITLE
[SPARK-31321][SQL] Remove SaveMode check in v2 FileWriteBuilder

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

The `SaveMode` is resolved before we create `FileWriteBuilder` to build `BatchWrite`.

In https://github.com/apache/spark/pull/25876, we removed save mode for DSV2 from DataFrameWriter. So that the `mode` method is never used which makes `validateInputs` fail determinately without `mode` set.

### Why are the changes needed?
rm dead code. 

### Does this PR introduce any user-facing change?

no

### How was this patch tested?

existing tests.

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/FileWriteBuilder.scala [hunk 1]
```diff
@@ -46,12 +45,6 @@ abstract class FileWriteBuilder(
   private val schema = info.schema()
   private val queryId = info.queryId()
   private val options = info.options()
-  private var mode: SaveMode = _
-
-  def mode(mode: SaveMode): WriteBuilder = {
```
Comment (dongjoon-hyun) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/FileWriteBuilder.scala:51]: @yaooqinn . Technically, removal of dead code is not a bug fix. Please re-category this as an `Improvement`.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/FileWriteBuilder.scala [hunk 2]
```diff
@@ -68,26 +61,8 @@ abstract class FileWriteBuilder(
     lazy val description =
       createWriteJobDescription(sparkSession, hadoopConf, job, paths.head, options.asScala.toMap)
 
-    val fs = path.getFileSystem(hadoopConf)
-    mode match {
-      case SaveMode.ErrorIfExists if fs.exists(path) =>
```
Comment (dongjoon-hyun) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/FileWriteBuilder.scala:73]: Could you give me the pointer where handles this?

**Comment by dongjoon-hyun** [comment]
cc @cloud-fan

**Comment by cloud-fan** [comment]
I'm OK to remove dead code, this was probably leftover when we remove `SaveMode` from the v2 API. Can we make the PR description clearer?

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/FileWriteBuilder.scala (hunk 1 above)
Comment (yaooqinn) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/FileWriteBuilder.scala:51]: OK, thanks

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/FileWriteBuilder.scala (hunk 2 above)
Comment (yaooqinn) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/FileWriteBuilder.scala:73]: e.g. https://github.com/apache/spark/blob/8b01473e8bffe349b1ed993b61420d7d68896cd8/sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala#L603

**Comment by yaooqinn** [comment]
> I'm OK to remove dead code, this was probably leftover when we remove `SaveMode` from the v2 API. Can we make the PR description clearer?

thanks, referenced the related PR https://github.com/apache/spark/pull/25876 and updated the PR description

**Comment by cloud-fan** [comment]
thanks, merging to master/3.0!

**Comment by dongjoon-hyun** [comment]
Thank you, @cloud-fan and @yaooqinn .

**Pull request closed without merging**